### PR TITLE
Add Encoding module, with Encoder and LegacyEncoder classes.

### PR DESCRIPTION
**Pull request by jondeandres**

### CODE REVIEW

File: lib/rollbar/encoding/encoder.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      def initialize(object)
+        @object = object
+      end
+
+      def encode
+        value = object.to_s
+
+        encoded_value = force_encoding(value).encode(*encoding_args(value))
+
+        object.is_a?(Symbol) ? encoded_value.to_sym : encoded_value
```
Comment (coryvirok) [inline lib/rollbar/encoding/encoder.rb]: This may be a dumb question, but what is this line for? Why do we need to turn the string back into a symbol?

Comment (jondeandres) [inline lib/rollbar/encoding/encoder.rb]: mmmm, yah. We don't need that. I'll change it.

File: lib/rollbar/encoding.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    def self.encode(object)
+      can_be_encoded = object.is_a?(Symbol) || object.is_a?(String)
+
+      return object unless can_be_encoded
+
+      encoding_class.new(object).encode
+    end
+
+    def self.encoding_class
+      if String.instance_methods.include?(:encode)
```
Comment (joneslee85) [inline lib/rollbar/encoding.rb]: What do you think about moving `require` under this conditional block?

File: lib/rollbar/encoding/encoder.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+      def detect_encoding(v)
+        value = v.dup
+        ALL_ENCODINGS.detect do |encoding|
+          begin
+            value.force_encoding(encoding).encode(::Encoding::UTF_8).codepoints
+          rescue
+            false
+          else
+            true
```
Comment (joneslee85) [inline lib/rollbar/encoding/encoder.rb]: what do you think about moving this line behind line 36?

File: spec/rollbar/encoding/encoder_spec.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
+# encoding: utf-8
+require 'spec_helper'
+
+unless RUBY_VERSION.start_with?('1.8')
+  require 'rollbar/encoding/encoder' unless
```
Comment (joneslee85) [inline spec/rollbar/encoding/encoder_spec.rb]: I find the `unless` is a bit confusing.

Btw, I am wondering if you should unit test this Encoder class? What do you think about treating this class as private and test the Encoding class instead?

File: lib/rollbar/encoding.rb (hunk 2 above)
Comment (jondeandres) [inline lib/rollbar/encoding.rb]: I like that, thanks! I'll change it.

File: lib/rollbar/encoding/encoder.rb (hunk 3 above)
Comment (jondeandres) [inline lib/rollbar/encoding/encoder.rb]: Probably better, thanks.

File: spec/rollbar/encoding/encoder_spec.rb (hunk 4 above)
Comment (jondeandres) [inline spec/rollbar/encoding/encoder_spec.rb]: That `require` and `unless` is something weird. I'll review it.

@joneslee85 I understand your point, but I'd prefer to just test that class separately.

**Pull request merged**